### DISCUSSION TITLE
Better way to use bullmq for non-blocking background process

### DISCUSSION THREAD

**Discussion by elhananjair** [taskforcesh/bullmq#2469] [Q&A]
Hello friends, I am working on a web application using Sveltekit and using bullmq. The application uses `SMPP` protocol to send and receive SMS, and I am using bullmq mainly to do the sending SMS part as background service so that it won't block the front end while users use the web application. 
But I am not getting the result I am looking for, when the process of sending SMS starts (which triggers thousands of phone numbers). the front end freezes for some seconds until the process finishes, but I was expecting the front end would work without depending on what background service is running using bullmq. Let me share you some samples from my code: 

```
hooks.server.js

import { sendSMS} from '$lib/server/sendsms.js'
import { cleanupdb } from '$lib/server/dbcleanup.js'
import { building } from '$app/environment';
import { Queue, Worker } from 'bullmq';
import sql from "$lib/server/db";

if (!building) {
  
     checkSchedule().then(val => {
           //run sendingSMS function in case the system restarts and the time is greater than the scheduled time in bullmq repeat time
    })

}

    const send_sms_queue = new Queue('send_sms_queue', {
        connection: {
            host: 'localhost',
            port: 6379
        }
    });

    async function addJobs() {
        await send_sms_queue.add('sendSMS', {}, {
            repeat: {
  
                pattern: `${arrScheduledTime[2]} ${arrScheduledTime[1]} ${arrScheduledTime[0]} * * *`,  //this values retrieved from database based on user's setting
        });
    }
    
    const worker = new Worker('send_sms_queue', async () => {
         await sendSMS();
   }, {
           connection: {
               host: 'localhost',
               port: 6379
           }
       });
```

Is there anything I missing to make the front end not be blocked while the job is running in the background? 
Thank you.

**Comment by manast** [comment]
If "sendSMS" is very CPU intensive (although it seems like it should not be), then you can use sandboxed processors: https://docs.bullmq.io/guide/workers/sandboxed-processors
Then the process function will be run in a separate thread or process. However I would investigate why sendSMS is taking all the CPU, that seems strange to me...

**Comment by elhananjair** [comment]
@manast thank you for replying, 

Actually, sendSMS is not CPU CPU-intensive job but it takes around 20-30 seconds to finish sending SMS to about 10,000 numbers looping over phone numbers fetched from a database. 

In addition, I tried to make the call `async` just to make it at least non-blocking, but I will check and use `sandboxing processors` as you suggested.

**Reply by manast** [reply]
The duration of the process does not matter, and if the loop inside the processor does not keep the event loop busy it would not block the rest of the app. I would investigate as there may be a bug in the processor code.

**Reply by elhananjair** [reply]
Ok thank you
